### DISCUSSION TITLE
Send Base64 images?

### DISCUSSION THREAD

**Discussion by ministr-kaifa** [reo7sp/tgbot-cpp#263] [Q&A] [answered]
Is it possible to send images as base64?

**Comment by reo7sp** [comment, marked as answer]
Base64 is not a separate Telegram upload format. Decode it to bytes and pass those bytes to `InputFile::fromData(data, mimeType, fileName)`, then call `sendPhoto` or the appropriate send method. If you already have a Telegram `file_id`, pass that instead.
